packages/core/src/cloud/LlamaCloudIndex.ts

```diff
@@ -0,0 +1,38 @@
+import { BaseRetriever } from "../Retriever";
+import { RetrieverQueryEngine } from "../engines/query/RetrieverQueryEngine";
+import { BaseNodePostprocessor } from "../postprocessors";
+import { BaseSynthesizer } from "../synthesizers";
+import { BaseQueryEngine } from "../types";
+import { LlamaCloudRetriever, RetrieveParams } from "./LlamaCloudRetriever";
+import { CloudConstructorParams } from "./types";
+
+export class LlamaCloudIndex {
```

Comment:
**NOTE**: I intentionally didn't extend from `BaseIndex`. Reasons:
1. we don't need the complete implementation from `BaseIndex`
2. we could extract a common interface (containing `asRetriever` and `asQueryEngine`) but I doubt it will have a great value